KitePrintSDK/src/main/java/ly/kite/product/SingleCurrencyCost.java

```diff
@@ -0,0 +1,255 @@
+    {
+    mCurrency        = currency;
+    mAmount          = amount;
+    mFormattedAmount = formattedAmount;
+    }
+
+
+  public SingleCurrencyCost( Currency currency, BigDecimal amount )
+    {
+    this ( currency, amount, null );
+    }
+
+
+  // Constructor used by parcelable interface
+  private SingleCurrencyCost( Parcel parcel )
+    {
+    mCurrency        = (Currency)parcel.readSerializable();
+    mAmount          = (BigDecimal)parcel.readSerializable();
+    mFormattedAmount = parcel.readString();
+    }
+
+
+  ////////// Parcelable Method(s) //////////
+
+  /*****************************************************
+   *
+   * Describes the contents.
+   *
+   *****************************************************/
+  @Override
+  public int describeContents()
+    {
+    return ( 0 );
+    }
+
+
+  /*****************************************************
+   *
+   * Write the contents to a parcel.
+   *
+   *****************************************************/
+  @Override
+  public void writeToParcel( Parcel targetParcel, int flags )
+    {
+    targetParcel.writeSerializable( mCurrency );
+    targetParcel.writeSerializable( mAmount );
+    targetParcel.writeString( mFormattedAmount );
+    }
+
+
+  ////////// Method(s) //////////
+
+  /*****************************************************
+   *
+   * Returns the currency.
+   *
+   *****************************************************/
+  public Currency getCurrency()
+    {
+    return ( mCurrency );
+    }
+
+
+  /*****************************************************
+   *
+   * Returns the currency code.
+   *
+   *****************************************************/
+  public String getCurrencyCode()
+    {
+    return ( mCurrency.getCurrencyCode() );
+    }
+
+
+  /*****************************************************
+   *
+   * Returns the amount.
+   *
+   *****************************************************/
+  public BigDecimal getAmount()
+    {
+    return ( mAmount );
+    }
+
+
+  /*****************************************************
+   *
+   * Returns the amount as a double.
+   *
+   *****************************************************/
+  public double getAmountAsDouble()
+    {
+    return ( mAmount.doubleValue() );
+    }
+
+
+  /*****************************************************
+   *
+   * Returns the formatted.
+   *
+   *****************************************************/
+  public String getFormattedAmount()
+    {
+    return ( mFormattedAmount );
+    }
+
+
+  /*****************************************************
+   *
+   * Returns the amount as a displayable string for the
+   * supplied locale.
+   *
+   * If the currency matches the main currency for the
+   * supplied locale, then we use the number formatter
+   * to format the amount.
+   *
+   * If the currency is different, then we format
+   * the amount with the full currency code. We do this to
+   * avoid any ambiguity. For example, if we were to live in
```

Comment:
Ah nice, we should do the same on iOS as I suspect this ambiguity is handled correctly.